src/GraphQL/Utilities/NameValidator.cs

```diff
@@ -3,12 +3,52 @@
 
 namespace GraphQL.Utilities
 {
+    /// <summary>
+    /// Validator for GraphQL names.
+    /// </summary>
     public static class NameValidator
     {
         private const string RESERVED_PREFIX = "__";
         private const string NAME_RX = @"^[_A-Za-z][_0-9A-Za-z]*$";
 
-        public static void ValidateName(string name, string type = "field")
+        /// <summary>
+        /// Gets or sets current validation delegate. By default this delegate validates all names according
+        /// to the GraphQL <see href="http://spec.graphql.org/June2018/#sec-Names">specification</see>.
+        /// <br/>
+        /// Setting this delegate allows you to use names not conforming to the specification, for example
+        /// 'enum-member'. Only change it when absolutely necessary.
+        /// </summary>
+        public static Action<string, string> Validation = ValidateDefault;
+
+        /// <summary>
+        /// Gets or sets current validation delegate during schema initialization. By default this delegate
+        /// validates all names according to the GraphQL <see href="http://spec.graphql.org/June2018/#sec-Names">specification</see>.
+        /// <br/>
+        /// Setting this delegate allows you to use names not conforming to the specification, for example
+        /// 'enum-member'. Only change it when absolutely necessary.
+        /// </summary>
+        public static Action<string, string> ValidationOnSchemaInitialize = ValidateDefault;
+
+        /// <summary>
+        /// Validates a specified name.
+        /// </summary>
+        /// <param name="name">GraphQL name.</param>
+        /// <param name="type">Type of element: field, type, argument, enum.</param>
+        public static void ValidateName(string name, string type = "field") => Validation(name, type);
+
+        /// <summary>
+        /// Validates a specified name during schema initialization.
+        /// </summary>
+        /// <param name="name">GraphQL name.</param>
+        /// <param name="type">Type of element: field, type, argument, enum.</param>
+        public static void ValidateNameOnSchemaInitialize(string name, string type = "field") => ValidationOnSchemaInitialize(name, type);
+
+        /// <summary>
+        /// Validates a specified name according to the GraphQL <see href="http://spec.graphql.org/June2018/#sec-Names">specification</see>.
+        /// </summary>
+        /// <param name="name">GraphQL name.</param>
+        /// <param name="type">Type of element: field, type, argument, enum.</param>
+        public static void ValidateDefault(string name, string type)
```

Comment:
Perhaps `ValidateDefault` could be named something better -- it seems confusing -- but I don't have a better suggestion, and you have clear xml comments.